engine/websocketroutine_manager.go

```diff
@@ -334,3 +343,20 @@ func (m *websocketRoutineManager) printAccountHoldingsChangeSummary(o account.Ch
 		o.Amount,
 		o.Account)
 }
+
+// registerInterceptor registers an externally (GCT Library) defined dedicated
+// filter to intercept specific data types for strategy use.
+func (m *websocketRoutineManager) registerInterceptor(fn Interceptor) error {
+	if m == nil {
+		return fmt.Errorf("%T %w", m, ErrNilSubsystem)
+	}
+
+	if fn == nil {
+		return errNilInterceptorFunction
+	}
+
+	m.mu.Lock()
+	m.interceptor = fn
```

Comment:
What if there already has been an interceptor registered? It should be an error and you should also include a function to deregister